src/components/map/map.tsx

```diff
@@ -1,20 +1,27 @@
-import React, { useRef } from "react";
 import mapboxgl from "mapbox-gl";
+import React, { useEffect, useRef } from "react";
 import { useMapSetup } from "./hooks/use-map-setup";
+import { useMapStore } from "./map-store";
 
 mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_API_KEY;
 
 const Map: React.FC = () => {
   const mapContainer = useRef<HTMLDivElement | null>(null);
   const map = useRef<mapboxgl.Map | null>(null);
+  const { setMap } = useMapStore();
 
   useMapSetup(mapContainer, map);
 
-  return (
-    <>
-      <div ref={mapContainer} className="h-screen" />
-    </>
-  );
+  useEffect(() => {
+    if (map && map.current) {
+      setMap(map.current);
+    }
+    return () => {
+      setMap(undefined);
+    };
+  }, [map]);
```

Comment:
could we restructure this file a bit and move all the setup stuff into the `useMapSetup` hook?
like we define the ref here, hand it over to the hook, then call `setMap` in another `useEffect` hook. Is there an advantage of having the declaration of the `ref` and the `useEffect` calling `setMap` in this component? if not i'd suggest to move it into the `useMapSetup` hook

Reply:
good idea, will do